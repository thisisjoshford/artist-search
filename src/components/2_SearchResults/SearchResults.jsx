import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import withList from '../withList';

const Artist = ({ name, info, id }) => (
  <>
    <Link to={`/${name}/${id}`} >
      <p>{name}</p>
    </Link>
    <p>Info: {info}</p>
  </>
);

const SearchResults = withList(Artist, 'results'); 

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default SearchResults;
