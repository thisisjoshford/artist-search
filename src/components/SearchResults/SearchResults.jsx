import React from 'react';
import PropTypes from 'prop-types';
import withList from '../withList';

const Artist = ({ name }) => (
  <>
    <p>{name}</p>
  </>
);

const SearchResults = withList(Artist, 'results'); 

Artist.propTypes = {
  name: PropTypes.string.isRequired,
};

export default SearchResults;

// import React from 'react';
// import PropTypes from 'prop-types';

// export default function ArtistResults({ results }) {
//   console.log(results);

//   const listItems = results.map(item => (
//     <li key={item.id}>
//       <p>{item.name}</p>
//     </li>
//   ));

//   return (
//     <ul>
//       {listItems}
//     </ul>
//   );
// }

// ArtistResults.propTypes = {
//   results: PropTypes.array.isRequired,
// };
