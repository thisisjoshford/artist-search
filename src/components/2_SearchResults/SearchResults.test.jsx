import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults';

describe('Search component', () => {
  it('renders Search', () => {
    const artists = [
      {
        id: '1234',
        name: 'Tool',
        info: 'Real Tool Band'
      },
      {
        id: '12345',
        name: 'Tool',
        info: 'not so real Tool band'
      }
    ];
    const wrapper = shallow(
      <SearchResults results={artists}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
