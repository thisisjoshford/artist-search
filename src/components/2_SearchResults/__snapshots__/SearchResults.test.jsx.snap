// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<ul>
  <li
    key="1234"
  >
    <Artist
      id="1234"
      info="Real Tool Band"
      name="Tool"
    />
  </li>
  <li
    key="12345"
  >
    <Artist
      id="12345"
      info="not so real Tool band"
      name="Tool"
    />
  </li>
</ul>
`;
